pkgs/by-name/an/antora-ui-default/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  fetchFromGitLab,
+  lib,
+  stdenvNoCC,
+}:
+let
+  srcFetchFromGitLab = {
+    hash = "sha256-q2FwkwzjanxTIxjMpCyMpzPt782uYZiWVdZ7Eev79oM=";
+    owner = "trueNAHO";
+    repo = "antora-ui-default";
+    rev = "83bf9bf5f22a6dee397f8b089eb0315c14a278b5";
+  };
+in
+stdenvNoCC.mkDerivation {
+  installPhase = ''install --mode 755 -D "$src/ui-bundle.zip" "$out"'';
+
+  meta = {
+    description = "Antora default UI bundle";
+    homepage = "https://gitlab.com/antora/antora-ui-default";
+    license = lib.licenses.mpl20;
+
+    longDescription = ''
+      > A UI bundle is a [ZIP
+      > archive](https://en.wikipedia.org/wiki/Zip_(file_format)) or directory
+      > that contains one or more UIs for a site.
+      >
+      > -- Antora
+      >    https://docs.antora.org/antora/3.1/playbook/ui-bundle-url
+
+      This UI bundle is intended to be passed to `antora`'s `--ui-bundle-url`
+      flag or injected into the [`ui.bundle.url`
+      key](https://docs.antora.org/antora/3.1/playbook/ui-bundle-url/#url-key)
+      to avoid irreproducible
+      [`https://gitlab.com/antora/antora-ui-default/-/jobs/artifacts/HEAD/raw/build/ui-bundle.zip?job=bundle-stable`](https://gitlab.com/${srcFetchFromGitLab.owner}/${srcFetchFromGitLab.repo}/-/blob/${srcFetchFromGitLab.rev}/README.adoc#user-content-use-the-default-ui)
+      references.
+    '';
+
+    maintainers = [ lib.maintainers.naho ];
+    platforms = lib.platforms.all;
+  };
+
+  phases = [ "installPhase" ];
+  pname = srcFetchFromGitLab.repo;
+
+  # The UI bundle is fetched from lib.maintainers.naho's antora-ui-default fork
+  # for the following reasons:
+  #
+  # > The UI bundle is currently unpackaged [1] [2], and only accessible by
+  # > fetching the latest GitLab artifact or building from source. Neither
+  # > method is reliably reproducible, as artifacts are deleted over time and
+  # > building from source requires insecure Node 10.
+  # >
+  # > The solution is to version control the UI bundle.
+  # >
+  # > [...]
+  # >
+  # > [1]: https://gitlab.com/antora/antora-ui-default/-/issues/135
+  # > [2]: https://gitlab.com/antora/antora-ui-default/-/issues/211
+  # >
+  # > -- [3]
+  #
+  # To avoid bloating the repository archive size, the UI bundle is not stored
+  # in Nixpkgs.
+  #
+  # For reference, the UI bundle from [3] is 300K large.
+  #
+  # [3]: https://gitlab.com/trueNAHO/antora-ui-default/-/commit/83bf9bf5f22a6dee397f8b089eb0315c14a278b5
```

Comment:
This wants to be landed on top of the repo property

Reply:
> This wants to be landed on top of the repo property

What exactly do you mean by "landed on top of the repo property"?

Do you mean that the comment should be placed as follows:

```nix
src = fetchFromGitLab {
  hash = null;
  owner = null;

  # > The UI bundle is fetched from lib.maintainers.naho's antora-ui-default
  # > fork for the following reasons:
  # >
  # > [...]
  repo = null;

  rev = null;
};
```

Would it not be better to place the comment like this:

```nix
# > The UI bundle is fetched from lib.maintainers.naho's antora-ui-default fork
# > for the following reasons:
# >
# > [...]
src = fetchFromGitLab {
  hash = null;
  owner = null;
  repo = null;
  rev = null;
};
```

Reply:
I'm fine with either, though I did mean the first example **and** I misspoke about the exact field, intending `owner` not `repo` 😳 

I meant this since the rest of the fields (hash, repo, rev) follow as a natural consequence of picking the owner.

But landing it on top of the src is fine with me.

Reply:
Alright. I will just keep it on top of `src` then.